server/application/application.go

```diff
@@ -334,24 +334,69 @@ func (s *Server) PodLogs(q *PodLogsQuery, ws ApplicationService_PodLogsServer) e
 
 // Sync syncs an application to its target state
 func (s *Server) Sync(ctx context.Context, syncReq *ApplicationSyncRequest) (*ApplicationSyncResult, error) {
-	log.Infof("Syncing application %s", syncReq.Name)
-	app, err := s.Get(ctx, &ApplicationQuery{Name: syncReq.Name})
+	return s.deployAndPersistDeploymentInfo(ctx, syncReq.Name, syncReq.Revision, nil, syncReq.DryRun)
+}
+
+func (s *Server) Rollback(ctx context.Context, rollbackReq *ApplicationRollbackRequest) (*ApplicationSyncResult, error) {
+	app, err := s.Get(ctx, &ApplicationQuery{Name: rollbackReq.Name})
+	if err != nil {
+		return nil, err
+	}
+	var deploymentInfo *appv1.DeploymentInfo
+	for _, info := range app.Status.RecentDeployments {
+		if info.Id == rollbackReq.Id {
+			deploymentInfo = &info
+			break
+		}
+	}
+	if deploymentInfo == nil {
+		return nil, fmt.Errorf("application %s does not have deployment with id %v", rollbackReq.Name, rollbackReq.Id)
```

Comment:
Instead of fmt.Errorf, can we use `status.Errorf(codes.InvalidArgument`

Reply:
done. thank you!